### PR TITLE
Scientist backend

### PR DESCRIPTION
**Pull request by hannagabor**

### CODE REVIEW

File: backend/heroes.py [hunk 1]
```diff
@@ -82,15 +86,19 @@ def before_step(self):
 
   def after_step(self):
     pass
-  
+
   def step(self, state, step_number):
+    if self.is_frozen():
+      return
```
Comment (darabos) [inline backend/heroes.py:88]: 👍

File: backend/heroes.py [hunk 2]
```diff
@@ -299,3 +307,27 @@ def move(self, state):
...
+      'name': 'Teacher',
+      'description':
+      '''Derek gains inspiration from teaching. Every time he attacks a different
+opponent, his inspiration increases.''',
+      'unlockLevel': 1,
+    },
+    {
+      'name': "Aumann's agreement theorem",
+      'description':
+      '''When Derek meets another rationalist, they argue until one of them changes his oppinion.'''
```
Comment (darabos) [inline backend/heroes.py]:
```suggestion
      '''When Derek meets another rationalist, they argue until one of them changes his opinion.'''
```

I've been thinking about phrasing these abilities. I think I also went too heavy on style to the detriment of explaining what actually happens. Our visual indicators are way too weak to hope that the player will see what happens. This text is all they have to go on. I think we must include a clear technical description.

File: backend/heroes.py [hunk 3]
```diff
@@ -299,3 +303,27 @@ def move(self, state):
...
+  shape = shapes.scientist
+  in_conversation_with = None
+  influence_per_level = 0.2
+
+  abilities = [
+    {
+      'name': 'Teacher',
+      'description':
+      '''Derek gains inspiration from teaching. Every time he attacks a different
+opponent, his inspiration increases.''',
```
Comment (darabos) [inline backend/heroes.py:319]: Like this one is perfect.

**Comment by hannagabor** [comment]
Thanks for the ideas!

**Pull request merged**